src/core/core/rpc_engine.cpp

```diff
@@ -456,9 +456,10 @@ namespace dsn {
     rpc_request_task* rpc_server_dispatcher::on_request(message_ex* msg, service_node* node)
     {
         rpc_handler_info* handler = nullptr;
-        if (msg->header->rpc_name_fast.local_binary_hash == message_ex::s_local_binary_hash)
+        auto binary_hash = msg->header->rpc_name_fast.local_hash;
+        if (binary_hash == message_ex::s_local_hash && binary_hash != 0)
```

Comment:
I actually firstly did as you suggested:-) The reason to put it later is that between replica servers, we will use this optimization a lot. For client requests, we may not use this as the client-lib may run inside 3rd party apps. So it really depends. As long as we have s-local-hash in replica server non-zero, the first condition also applies for the 2nd scenario.